packages/design-toolkit/src/components/slider/index.tsx

```diff
@@ -26,7 +26,176 @@ import {
 import { Tooltip } from '../tooltip';
 import { TooltipTrigger } from '../tooltip/trigger';
 import styles from './styles.module.css';
-import type { SliderProps } from './types';
+import type { SliderMarker, SliderMarkersConfig, SliderProps } from './types';
+
+/**
+ * Normalizes the markers configuration into a consistent array format
+ */
+function normalizeMarkers(
+  markersConfig: SliderMarkersConfig | undefined,
+  min: number,
+  max: number,
+): SliderMarker[] {
+  if (markersConfig === undefined) {
+    return [];
+  }
+
+  // Number of evenly spaced markers
+  if (typeof markersConfig === 'number') {
+    if (markersConfig < 2) {
+      return [];
+    }
+    const step = (max - min) / (markersConfig - 1);
+    return Array.from({ length: markersConfig }, (_, i) => ({
+      value: min + step * i,
+    }));
+  }
+
+  // Array of numbers or marker objects
+  if (Array.isArray(markersConfig)) {
+    return markersConfig.map((item) =>
+      typeof item === 'number' ? { value: item } : item,
+    );
+  }
+
+  return [];
+}
+
+/**
+ * Calculates the GCD of two numbers with floating point precision
+ */
+function gcd(num1: number, num2: number): number {
```

Comment:
Something like this should either be in a DevTK package or pulled from an existing dependency. It's too generic and too common to be inlined here